xbmc/addons/Repository.cpp

```diff
@@ -115,9 +115,16 @@ CRepository::CRepository(const AddonInfoPtr& addonInfo)
         (dir.maxversion.empty() || version <= dir.maxversion))
       m_dirs.push_back(std::move(dir));
   }
+
+  // old (dharma compatible) way of defining the addon repository structure, is no longer supported
+  // we error out so the user knows how to migrate. The <dir> way is supported since gotham.
+  //! @todo remove if block completely in v21
   if (!Type(ADDON_REPOSITORY)->GetValue("info").empty())
   {
-    m_dirs.push_back(ParseDirConfiguration(*Type(ADDON_REPOSITORY)));
+    CLog::Log(LOGERROR,
+              "Repository add-on {} uses old schema definition for the repository extension point! "
+              "This is no longer supported, please update your addon to use <dir> definitions.",
+              ID());
   }
 
   for (auto const& dir : m_dirs)
```

Comment:
maybe log a warning/error here if m_dirs.empty() .. shouldnt happen but in case it does we end up with an invalid CRepository from the ctor?

Reply:
yeah nice point. It really doesn't do any harm but it will simply do nothing even though its still listed. Will add an error log entry